internal/blockchain/thresholdstate.go

```diff
@@ -470,34 +461,23 @@ func (b *BlockChain) nextThresholdState(version uint32, prevNode *blockNode, che
 	return stateTuple, nil
 }
 
-// deploymentState returns the current rule change threshold for a given stake
-// version and deploymentID.  The threshold is evaluated from the point of view
-// of the block node passed in as the first argument to this method.
+// deploymentState returns the current rule change threshold for a given
+// deployment.  The threshold is evaluated from the point of view of the block
+// node passed in as the first argument to this method.
 //
 // It is important to note that, as the variable name indicates, this function
 // expects the block node prior to the block for which the deployment state is
 // desired.  In other words, the returned deployment state is for the block
 // AFTER the passed node.
 //
 // This function MUST be called with the chain state lock held (for writes).
-func (b *BlockChain) deploymentState(prevNode *blockNode, version uint32, deploymentID string) (ThresholdStateTuple, error) {
-	for k := range b.chainParams.Deployments[version] {
-		if b.chainParams.Deployments[version][k].Vote.Id == deploymentID {
-			checker := deploymentChecker{
-				deployment: &b.chainParams.Deployments[version][k],
-				chain:      b,
-			}
-			cache := &b.deploymentCaches[version][k]
-			return b.nextThresholdState(version, prevNode, checker, cache)
-		}
-	}
-
-	invalidState := ThresholdStateTuple{
-		State:  ThresholdInvalid,
-		Choice: invalidChoice,
+func (b *BlockChain) deploymentState(prevNode *blockNode, deployment *deploymentInfo) (ThresholdStateTuple, error) {
+	checker := deploymentChecker{
+		deployment: deployment.deployment,
+		chain:      b,
 	}
-	str := fmt.Sprintf("deployment ID %s does not exist", deploymentID)
-	return invalidState, contextError(ErrUnknownDeploymentID, str)
+	cache := deployment.cache
+	return b.nextThresholdState(deployment.version, prevNode, checker, cache)
```

Comment:
The funcs `nextThresholdState` and `stateLastChanged` could have their signatures reduced a bit by passing in the whole `deploymentInfo` struct rather than `deploymentInfo.version` and `deployment.cache` separately.

Reply:
Good point.  I was mostly trying to reduce the number of changes to ease review, but I agree that it ultimately ends up cleaner, so I've updated this accordingly.